alerter-component/src/main/java/io/github/steliospaps/ighackathon/instrumentpricealerter/alertercomponent/websocket/client/WebsocketClient.java

```diff
@@ -147,6 +147,9 @@ public void shutdown() {
 			case "Quote":
 				publishPrices(mapper.treeToValue(jsonNode, Quote.class));
 				return Flux.empty();
+			case "NegotiationReject":
+				log.warn("Session failed to logon, reason={}", jsonNode.get("Reason"));
+				return Flux.empty();
```

Comment:
does it make it exit? or we rely on the timer to kick us out?

Reply:
no this doesnt make it exit, it handles this like heartbeat messages. just returns an empty flux (no elements to send back) in response to the NegotitationReject. The server kicks out a client immediately after this reject message.